src/platform.rs

```diff
@@ -351,6 +354,39 @@ pub fn sse41_detected() -> bool {
     false
 }
 
+#[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
+#[inline(always)]
+pub fn neon_detected() -> bool {
+    // A testing-only short-circuit.
+    if cfg!(feature = "no_neon") {
+        return false;
+    }
+    #[cfg(target_arch = "aarch64")]
+    {
+        return true
+    }
+    // Static check, e.g. for building with target-cpu=native.
+    // ... but rust does not provide this
+    //#[cfg(target_feature = "armv7-a+simd")]
+    //{
+    //    return true;
+    //}
+    // Dynamic check
+    #[cfg(all(unix, target_arch = "arm"))]
+    {
+    // https://doc.rust-lang.org/beta/std/macro.is_arm_feature_detected.html
+    // should work, and should be available and should be using
+    // getauxval, yet it's not available. So use libc ffi bindings to
+    // get to getauxval ourselves.
+    // #define HWCAP_ARM_VFP 64
+    // #define HWCAP_ARM_NEON 4096
+    //
+        unsafe {
+            return (libc::getauxval(libc::AT_HWCAP) & (64 | 4096)) != 0
+        }
+    }
+}
```

Comment:
This implements neon_detect the same way as done in the C implementation.

Note this is my first time writing rust code. So please review this carefully.

Namely it's simply true on aarch64, and on armhf an unsafe call to libc getauxval is made. It is implemented by Android's bionic library & glibc.

I have slight concerns that it might not be portable to FreeBSD/MacOS. However, I don't think there are any 32-bit ARM MacOS things left. But FreeBSD is a concern.

For example, I have no idea how above behaves if `AT_HWCAP` is not defined, or `getauxval` is not defined in the target libc. I hope that compilation fails, when building against such libc.